tools/converter/mule2ballerina/src/main/java/org/wso2/ei/tools/mule2ballerina/builder/BallerinaASTModelBuilder.java

```diff
@@ -0,0 +1,228 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.ei.tools.mule2ballerina.builder;
+
+import org.ballerinalang.model.BLangPackage;
+import org.ballerinalang.model.BLangProgram;
+import org.ballerinalang.model.BallerinaFile;
+import org.ballerinalang.model.GlobalScope;
+import org.ballerinalang.model.NativeScope;
+import org.ballerinalang.model.builder.BLangModelBuilder;
+import org.ballerinalang.model.types.SimpleTypeName;
+import org.ballerinalang.util.program.BLangPackages;
+import org.ballerinalang.util.program.BLangPrograms;
+import org.ballerinalang.util.repository.PackageRepository;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Stack;
+
+/**
+ * {@code BallerinaASTModelBuilder} is a high level API to build ballerina AST from BLangModelBuilder class
+ */
+public class BallerinaASTModelBuilder {
+
+    private boolean processingActionInvocationStmt = false;
+    private Stack<BLangModelBuilder.NameReference> nameReferenceStack = new Stack<>();
+    private Stack<SimpleTypeName> typeNameStack = new Stack<>();
+    private BLangProgram programScope;
+    private BLangPackage bLangPackage;
+    private BLangModelBuilder modelBuilder;
+    private Map<String, String> ballerinaPackageMap = new HashMap<String, String>();
+
+    private static final PackageRepository PACKAGE_REPOSITORY = new PackageRepository() {
+        @Override
+        public PackageSource loadPackage(Path path) {
+            return null;
+        }
+
+        @Override
+        public PackageSource loadFile(Path path) {
+            return null;
+        }
+    };
+
+    public BallerinaASTModelBuilder() {
+
+        GlobalScope globalScope = BLangPrograms.populateGlobalScope();
+        NativeScope nativeScope = BLangPrograms.populateNativeScope();
+
+        programScope = new BLangProgram(globalScope, nativeScope, BLangProgram.Category.SERVICE_PROGRAM);
+        bLangPackage = new BLangPackage(".", PACKAGE_REPOSITORY, programScope);
+        BLangPackage.PackageBuilder packageBuilder = new BLangPackage.PackageBuilder(bLangPackage);
+        modelBuilder = new BLangModelBuilder(packageBuilder, ".");
+
+        String[] packages = BLangPackages.getBuiltinPackageNames();
+        for (String aPackage : packages) {
+            String[] bits = aPackage.split("\\.");
+            String pkgName = bits[bits.length - 1];
+            ballerinaPackageMap.put(pkgName, aPackage);
+        }
+    }
+
+    public void addImportPackage(String pkgPath, String asPkgName) {
```

Comment:
Can you add doc comments to the public methods?

Reply:
They've been added in a later commit.